### DISCUSSION TITLE
Run java program on wolfi base image

### DISCUSSION THREAD

**Discussion by omkarkakade2017** [wolfi-dev/os#17981] [Q&A] [answered]
Hello Team,
I am looking for running java program over wolfi base image by installing Openjdk. Can anyone help on this?
My docker file is
`FROM cgr.dev/chainguard/wolfi-base

WORKDIR /app

RUN apk update && apk add openjdk-21

COPY HelloWolfi.class /app

ENTRYPOINT ["java", "HelloWolfi"]

Any example or tutorial will also helpful.
`

**Comment by amouat** [comment, marked as answer]
Hi @omkarkakade2017.

Are you having a problem with the image? It's not really clear to me what you're asking. We do have some Java resources though:

 - https://edu.chainguard.dev/chainguard/chainguard-images/videos/java-images/
 - https://www.crowdcast.io/c/learninglabs-mar24

There is also a supporting GitHub repo you can play with: https://github.com/chainguard-dev/learning-labs-java
